### ISSUE TITLE
Find-PSResource does not show all prerelease versions when using wildcard *

### ISSUE THREAD

**Issue by ThomasNieto** [PowerShell/PSResourceGet#685]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PowerShellGet)
- [X] Search the existing issues.

### Steps to reproduce

If you run the following command to show all versions of PowerShellGet on the repository you'll notice that version `3.0.0.0-beta1` is missing from the list however if you search for it directly it shows up and on [PowerShellGallery](https://www.powershellgallery.com/packages/PowerShellGet/3.0.0-beta1). It looks to be only returning the latest prerelease or stable version for that specific version number. You can also see this more versions with Pester.

### Expected behavior

```console
PS C:\> Find-PSResource -Name PowerShellGet -Version * -Prerelease

Name          Version  Prerelease Repository Description
----          -------  ---------- ---------- -----------
PowerShellGet 3.0.14.0 beta14     PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 3.0.12.0 beta       PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 3.0.11.0 beta       PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 3.0.0.0  beta10     PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 3.0.0.0  beta1      PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.5.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.4.1             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.4.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.3.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.2.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.1.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.5.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.4.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.3.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.2.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.1.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.0.4.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.0.3.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.0.1.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.0.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.6.7.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.6.6.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.6.5.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.6.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.5.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.3.2             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.3.1             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.2.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.1.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…

PS C:\> Find-PSResource -Name PowerShellGet -Version 3.0.0.0-beta1 -Prerelease

Name          Version Prerelease Repository Description
----          ------- ---------- ---------- -----------
PowerShellGet 3.0.0.0 beta1      PSGallery  PowerShell module with commands for discovering, installing, updating and …
```

### Actual behavior

```console
PS C:\> Find-PSResource -Name PowerShellGet -Version * -Prerelease

Name          Version  Prerelease Repository Description
----          -------  ---------- ---------- -----------
PowerShellGet 3.0.14.0 beta14     PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 3.0.12.0 beta       PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 3.0.11.0 beta       PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 3.0.0.0  beta10     PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.5.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.4.1             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.4.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.3.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.2.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.1.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.2.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.5.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.4.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.3.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.2.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.1.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.1.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.0.4.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.0.3.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.0.1.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 2.0.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.6.7.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.6.6.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.6.5.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.6.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.5.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.3.2             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.3.1             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.2.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.1.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…
PowerShellGet 1.1.0.0             PSGallery  PowerShell module with commands for discovering, installing, updating and…

PS C:\> Find-PSResource -Name PowerShellGet -Version 3.0.0.0-beta1 -Prerelease

Name          Version Prerelease Repository Description
----          ------- ---------- ---------- -----------
PowerShellGet 3.0.0.0 beta1      PSGallery  PowerShell module with commands for discovering, installing, updating and …
```

### Error details

```console
N/A
```

### Environment data

```powershell
Name          Version
----          -------
PowerShellGet 3.0.14

Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by SydneyhSmith** [comment]
With beta20 it appears to be returning unlisted versions as well 
```
Find-PSResource -Name PowerShellGet -Version * -Prerelease -Repository PSGallery

Name          Version Prerelease Repository Description
----          ------- ---------- ---------- -----------
PowerShellGet 3.0.19  beta19     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.18  beta18     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.17  beta17     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.16  beta16     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.15  beta15     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.14  beta14     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.13  beta       PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.12  beta       PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.11  beta       PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test9      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test8      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test7      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test6      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test5      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test4      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test3      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test2      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test156    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test155    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test154    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test153    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test152    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test151    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test150    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test149    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test148    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test144    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test143    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test142    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test141    PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test14     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test13     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test12     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test11     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test10     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   test1      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta9      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta8      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta7      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta6      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta5      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta4      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta3      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta2      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta10     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta1      PSGallery  PowerShell module with commands for discovering…
PowerShellGet 3.0.0   beta08     PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.2.5.1            PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.2.5              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.2.4.1            PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.2.4              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.2.3              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.2.2              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.2.1              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.2                PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.1.5              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.1.4              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.1.3              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.1.2              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.1.1              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.1.0              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.0.4              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.0.3              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.0.2              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.0.1              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 2.0.0              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.6.7              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.6.6              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.6.5              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.6.0              PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.5.0.0            PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.1.3.2            PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.1.3.1            PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.1.2.0            PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.1.1.0            PSGallery  PowerShell module with commands for discovering…
PowerShellGet 1.1.0.0            PSGallery  PowerShell module with commands for discovering…
```

**Comment by alerickson** [comment]
Closing this issue since the original issue is resolved as of 3.0.21-beta21, but opened issue #1118 to track second bug.
